### ISSUE TITLE
[Request] Add support to TSOL-MX*

### ISSUE THREAD

**Issue by liberodark** [tbnobody/OpenDTU#1931]
### Is your feature request related to a problem? Please describe.

Hi,

Can you add support to TSOL-MX400 TSOL-MX450 TSOL-MX500.

Best Regards

### Describe the solution you'd like

I'm going to purchase play 2 with a SOL-MX450.
So I could perhaps help with the tests also I have an openDTU.

**Comment by Franckled1** [comment]
Hello,
I am in the same configuration than liberodark.
Can you add TSOL MX450 to current inverters ?
Have a good day

**Comment by stefan123t** [comment]
According to the datasheet that @liberodark linked the TSUN TSOL-MX series models have a Communication Module: WiFi.

I assume the Gen3 here does not mean it is using Hoymiles Gen3 RF Protocol but rather that it is similar to the HMS-WiFi series models with an integrated DTU-BI (built-in).

You may refer to the respective projects for mimicking an S-Miles Installer App / S-Miles Cloud from @DennisOSRM and @rovo89 in their own repositories.

**Comment by stefan123t** [comment]
Yes the PR #3052 from @peterfroehlich and the other one downstream from @henkwiedig that @ruckerzerg has mentioned may well support the TSUN/TSOL WiFi models too.

I can not confirm or test this due to a lack of those DTU-BI devices in my household. But I would suggest you give it a try, if at all, it may require some minor changes eg. to whitelist the TSUN/TSOL WiFi model serial ids and map them to the right data model.
